lib/std/debug.zig

```diff
@@ -50,163 +40,378 @@ pub const LineInfo = struct {
+
+        var buf: [mem.page_size]u8 = undefined;
+        var line: usize = 1;
+        var column: usize = 1;
+        var abs_index: usize = 0;
+        while (true) {
+            const amt_read = try f.read(buf[0..]);
+            const slice = buf[0..amt_read];
+
+            for (slice) |byte| {
+                if (line == line_info.line) {
+                    try writer.writeByte(byte);
+                    if (byte == '\n') {
+                        return;
+                    }
+                }
+                if (byte == '\n') {
+                    line += 1;
+                    column = 1;
+                } else {
+                    column += 1;
+                }
+            }
+
+            if (amt_read < buf.len) return error.EndOfFile;
+        }
+    }
+
+    pub fn attemptWriteLineFromSourceFile(writer: anytype, line_info: LineInfo) !bool {
+        writeLineFromFileAnyOs(writer, line_info) catch |err| {
+            switch (err) {
+                error.EndOfFile, error.FileNotFound => {},
+                error.BadPathName => {},
+                error.AccessDenied => {},
+                else => return err,
+            }
+            return false;
+        };
+        return true;
+    }
+
+    /// When overriding panicTerminate, if you *must* lock a mutex, be careful not to panic!
+    /// Otherwise, this will would call panicTerminate recursively and deadlock.
+    pub const panicTerminate = os.abort;
+};
+
+const is_default_config = @hasDecl(root, "debug_config");
+const config = if (is_default_config)
+    root.debug_config
+else
+    default_config;
+
+const has_writer_decl = @hasDecl(config, "getWriter");
+
+/// The function used to get a writer for debugging.
+/// Slightly different name than in config to avoid redefinition in default.
+const getWrit = if (has_writer_decl)
+    config.getWriter
+else
+    default_config.getWriter;
+
+/// The function used to determine which escape codes can be used.
+/// Slightly different name than in config to avoid redefinition in default.
+const getTTYConfig = if (@hasDecl(config, "detectTTYConfig"))
+    config.detectTTYConfig
+else if (has_writer_decl)
+    @compileError("getWriter exists in config, so detectTTYConfig must also exist")
+else
+    default_config.detectTTYConfig;
+
+const has_address_symbol_mapping_decl = @hasDecl(config, "AddressSymbolMapping");
+const using_default_address_symbol_mapping = is_default_config or
+    !has_address_symbol_mapping_decl;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const AddrSymMapping = if (has_address_symbol_mapping_decl)
+    config.AddressSymbolMapping
+else
+    default_config.AddressSymbolMapping;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const writeLineFromSourceFile = if (@hasDecl(config, "attemptWriteLineFromSourceFile"))
+    config.attemptWriteLineFromSourceFile
+else
+    default_config.attemptWriteLineFromSourceFile;
```

Comment:
At the moment, the default config is placed inside `debug.zig`. So we have to do dumb stuff like this instead of just using the same name everywhere.

Approaches:
 - Eat it and use a different name (this done right now)
 - Move the default config to a separate file.
 - Implement https://github.com/ziglang/zig/issues/678 : )

Reply:
Based on a conversation on discord, it looks like this will be implemented in stage 2. So for now we will just eat it.